src/Sarif/EnumeratedArtifact.cs

```diff
@@ -16,44 +16,162 @@ public EnumeratedArtifact(IFileSystem fileSystem)
             FileSystem = fileSystem;
         }
 
+        internal byte[] bytes;
         internal string contents;
 
+        private bool? isBinary;
+        private Encoding encoding;
+
         public Uri Uri { get; set; }
 
+        public bool IsBinary
+        {
+            get
+            {
+                return this.isBinary ?? IsArtifactBinary();
+            }
+        }
+
+        internal bool IsArtifactBinary()
```

Comment:
Suggest: make private to narrow API surface to IsBinary?

Reply:
'internal' is still private, i tend to do this to make things testable but not public. As it happens I can now delete this helper entirely.

It now occurs to me: this new property value is untested!! Need to add it to the test validations. So I'm glad you pointed to it. :)